### PR TITLE
[Merged by Bors] - feat: 2-adic norm of the n-th harmonic number

### PR DESCRIPTION
**Pull request by kodyvajjha**
feat: 2-adic norm of the n-th harmonic number

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/Padics/Harmonic.lean [hunk 1]
```diff
@@ -49,6 +51,12 @@ theorem padicValRat_two_harmonic {n : ℕ} : padicValRat 2 (harmonic n) = -Nat.l
         padicValRat.inv, padicValRat.of_nat, min_neg_neg, neg_inj, ← Nat.cast_max, Nat.cast_inj]
     exact Nat.max_log_padicValNat_succ_eq_log_succ n
 
+/-- The 2-adic norm of the n-th harmonic number is 2 raised to the logarithm of n in base 2. -/
+lemma padicNorm_two_harmonic {n : ℕ} (hn : n ≠ 0) :
+    ‖(harmonic n : ℚ_[2])‖ = 2 ^ (Nat.log 2 n) := by
+  rw [padicNormE.eq_padicNorm, padicNorm.eq_zpow_of_nonzero (harmonic_pos hn).ne',
+  padicValRat_two_harmonic, neg_neg, zpow_coe_nat, Rat.cast_pow, Rat.cast_coe_nat, Nat.cast_ofNat]
```
Comment (alexjbest) [inline Mathlib/NumberTheory/Padics/Harmonic.lean:58]:
```suggestion
    padicValRat_two_harmonic, neg_neg, zpow_coe_nat, Rat.cast_pow, Rat.cast_coe_nat, Nat.cast_ofNat]
```

**Comment by alexjbest** [comment]
seems fine, what is the motivation for adding this though?

**Comment by alexjbest** [comment]
Ok, its just that we have by now several ways of expressing p-adic valuation / norm, and we probably don't want all statements for all versions of that, but this seems fine to me.

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**